airflow/providers/amazon/aws/sensors/s3_upload_session_complete.py

```diff
@@ -0,0 +1,172 @@
+                 verify: Optional[Union[bool, str]] = None,
+                 inactivity_period: float = 60 * 60,
+                 min_objects: int = 1,
+                 previous_objects: Optional[Set[str]] = None,
+                 allow_delete: bool = True,
+                 *args, **kwargs) -> None:
+
+        super().__init__(*args, **kwargs)
+
+        self.bucket = bucket_name
+        self.prefix = prefix
+        if inactivity_period < 0:
+            raise ValueError("inactivity_period must be non-negative")
+        self.inactivity_period = inactivity_period
+        self.min_objects = min_objects
+        self.previous_objects = previous_objects if previous_objects else set()
+        self.inactivity_seconds = 0
+        self.allow_delete = allow_delete
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.last_activity_time: Optional[datetime] = None
+
+    @cached_property
+    def hook(self):
+        """
+        Returns S3Hook.
+        """
+        return S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
+
+    def is_upload_session_complete(self, current_objects: Set[str]) -> bool:
+        """
+        Checks whether new objects have been uploaded and the inactivity_period
+        has passed and updates the state of the sensor accordingly.
+
+        :param current_objects: set of object ids in bucket during last poke.
+        :type current_objects: set[str]
+        """
+        current_num_objects = len(current_objects)
+        if current_objects > self.previous_objects:
+            # When new objects arrived, reset the inactivity_seconds
+            # and update previous_objects for the next poke.
+            self.log.info("New objects found at %s resetting last_activity_time.",
+                          os.path.join(self.bucket, self.prefix))
+            self.log.debug("New objects: %s",
+                           "\n".join(current_objects - self.previous_objects))
+            self.last_activity_time = datetime.now()
+            self.inactivity_seconds = 0
+            self.previous_objects = current_objects
+            return False
+
+        if self.previous_objects - current_objects:
+            # During the last poke interval objects were deleted.
+            if self.allow_delete:
+                deleted_objects = self.previous_objects - current_objects
+                self.previous_objects = current_objects
+                self.last_activity_time = datetime.now()
+                self.log.info(
+                    "Objects were deleted during the last poke interval. Updating the file counter and "
+                    "resetting last_activity_time:\n    %s",
+                    deleted_objects
+                )
+                return False
+
+            raise AirflowException(
+                """
+                Illegal behavior: objects were deleted in {} between pokes.
+                """.format(os.path.join(self.bucket, self.prefix))
+            )
```

Comment:
```suggestion
            raise AirflowException(
                'Illegal behavior: objects were deleted in'
                f'{os.path.join(self.bucket, self.prefix)} between pokes.'
            )
```

Reply:
For example here I think the use of `"""` will produce:
```
    Illegal behavior...
```
not sure if it is intended.

The suggestion changes it to:
```
```
Illegal behavior...
```
```